unitTests/testArraySlice.cpp

```diff
@@ -0,0 +1,112 @@
+
+TYPED_TEST_P( PermutedSliceTest, IsContiguous )
+{
+  using PERMUTATION = TypeParam;
+  constexpr int NDIM = getDimension( PERMUTATION {} );
+
+  // Create a permuted array and resize with nontrivial sizes
+  Array< int, NDIM, PERMUTATION, int > a;
+  std::array< int, NDIM > dims{};
+  for( int i = 0 ; i < NDIM ; ++i )
+  {
+    dims[i] = 2+i;
+  }
+  resizeArray( a, dims, camp::make_idx_seq_t< NDIM >{} );
+
+  // The only way to get a contiguous (NDIM-1)-dimenstional slice is by
+  // chopping along the largest-stride dimension (i.e. first in permutation)
+  bool const contiguous = RAJA::as_array< PERMUTATION >::get()[0] == 0;
+
+  // There's no need to test all slices, but we might as well
+  for( int i = 0 ; i < dims[0] ; ++i )
+  {
+    EXPECT_EQ( a[i].isContiguous(), contiguous );
+  }
+}
+
+REGISTER_TYPED_TEST_CASE_P( PermutedSliceTest, IsContiguous );
+
+using Permutations2D = ::testing::Types< RAJA::PERM_IJ,
+                                         RAJA::PERM_JI >;
+
+using Permutations3D = ::testing::Types< RAJA::PERM_IJK,
+                                         RAJA::PERM_JIK,
+                                         RAJA::PERM_IKJ,
+                                         RAJA::PERM_KIJ,
+                                         RAJA::PERM_KJI,
+                                         RAJA::PERM_JKI >;
+
+using Permutations4D = ::testing::Types< RAJA::PERM_IJKL,
+                                         RAJA::PERM_JIKL,
+                                         RAJA::PERM_IKJL,
+                                         RAJA::PERM_KIJL,
+                                         RAJA::PERM_JKIL,
+                                         RAJA::PERM_KJIL,
+                                         RAJA::PERM_IJLK,
+                                         RAJA::PERM_JILK,
+                                         RAJA::PERM_ILJK,
+                                         RAJA::PERM_LIJK,
+                                         RAJA::PERM_JLIK,
+                                         RAJA::PERM_LJIK,
+                                         RAJA::PERM_IKLJ,
+                                         RAJA::PERM_KILJ,
+                                         RAJA::PERM_ILKJ,
+                                         RAJA::PERM_LIKJ,
+                                         RAJA::PERM_KLIJ,
+                                         RAJA::PERM_LKIJ,
+                                         RAJA::PERM_JKLI,
+                                         RAJA::PERM_KJLI,
+                                         RAJA::PERM_JLKI,
+                                         RAJA::PERM_LJKI,
+                                         RAJA::PERM_KLJI,
+                                         RAJA::PERM_LKJI >;
+
+INSTANTIATE_TYPED_TEST_CASE_P( 2D, PermutedSliceTest, Permutations2D );
+INSTANTIATE_TYPED_TEST_CASE_P( 3D, PermutedSliceTest, Permutations3D );
+INSTANTIATE_TYPED_TEST_CASE_P( 4D, PermutedSliceTest, Permutations4D );
```

Comment:
Could you change these to just standard typed tests? There not a reason to use the parameter version here and that's what the other tests use (or should use).

Reply:
There is a somewhat weak reason. Because the following is not legal:
```
TYPED_TEST_CASE( PermutedSliceTest, Permutations2D );
TYPED_TEST_CASE( PermutedSliceTest, Permutations3D );
TYPED_TEST_CASE( PermutedSliceTest, Permutations4D );
```
It works if I put all permutations (2D/3D/4D) into a single type list, e.g.
```
using Permutations = ::testing::Types< RAJA::PERM_IJ, ..., ..., RAJA::PERM_LKJI >;
TYPED_TEST_CASE( PermutedSliceTest, Permutations );
```
which slightly loses on expressiveness, because the test names you see at runtime are something like `PermutedSliceTest/camp::idx_seq<0, 1, 2>/IsContiguous>`, so it's a bit more difficult to decipher the dimension, compared to having it right in the test name. Although it's a pretty minor detail.

In general I tend to prefer _type-parametrized_ tests to _typed_ tests, because they don't require pre-defining a global list of types to be tested. In cases like LAI this is the only option, since the list of types changes based on which packages you compile with, so a global type tuple is (nearly) impossible to define. In this case it doesn't really matter that much, so I'll do it.

Reply:
That's sounds reasonable to me, but I see you already made the change. Oh well.